webapp/src/ts/services/form.service.ts

```diff
@@ -326,7 +331,39 @@ export class FormService {
     }, null);
   }
 
-  async saveContact(form, docId, type, xmlVersion) {
+  async checkForDuplicates(doc, duplicateCheck, acknowledged) {
+    const parentId = doc ? doc.parent?._id : undefined;
+    const contactType = doc ? doc.contact_type ?? doc.type : undefined;
+    const siblings = await requestSiblings(this.dbService, parentId, contactType);
+    const expression = extractExpression(duplicateCheck);
+    const isCanonical = doc.is_canonical ? doc.is_canonical === 'true' : false;
+    acknowledged = acknowledged ?? false;
+
+    if (!isCanonical && expression && !acknowledged){
+      const duplicates = getDuplicates(
+        doc, 
+        siblings, 
+        {
+          expression, 
+          parseProvider: this.parseProvider, 
+          xmlFormsContextUtilsService: this.xmlFormsDuplicateUtilsService
+        }
+      );
+      return duplicates;
+    }
```

Comment:
nitpick: we can simplify/flatten this a bit by reversing the if-check:

```suggestion
    if (isCanonical || !expression || acknowledged) {
      return [];
    }
    return getDuplicates(
      doc, 
      siblings, 
      {
        expression, 
        parseProvider: this.parseProvider, 
        xmlFormsContextUtilsService: this.xmlFormsDuplicateUtilsService
      }
    );
```